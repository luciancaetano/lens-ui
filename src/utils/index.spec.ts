import {
  Layers, getPortalContainer, randomId, sleep,
} from '.';
import { CLASSES } from '..';

describe('utils', () => {
  it('Layers', () => {
    expect(Layers).toMatchSnapshot();
  });

  it('getPortalContainer', () => {
    const root = window.document.createElement('div');
    root.id = CLASSES.PortalRootContainer;
    window.document.body.appendChild(root);

    expect(getPortalContainer('my_portal')).toBeInTheDocument();
  });

  it('randomId', () => {
    expect(randomId('', 10, 'abcdef0123456789').length).toBe(10);
    expect(randomId('_prefix', 10, 'abcdef0123456789').startsWith('_prefix')).toBe(true);
  });

  it('sleep', async () => {
    const cancelRef = { current: null };

    const startDate = new Date();

    await sleep(1000, cancelRef);

    const diff = ((new Date()).getTime() - startDate.getTime());

    expect(diff >= 1000 && diff <= 1500).toBe(true);
  });

  it('sleep clear', async () => {
    const cancelRef = { current: null as any };
    const resolved = jest.fn();

    sleep(1000, cancelRef).then(resolved);

    if (typeof cancelRef.current === 'function') {
      cancelRef.current();
    }

    await sleep(100);

    expect(resolved).toHaveBeenCalled();
  });

  it('Layers', () => {
    expect(Layers).toMatchSnapshot();
  });
});
