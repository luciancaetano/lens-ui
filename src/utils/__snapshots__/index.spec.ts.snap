// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils Layers 1`] = `
Object {
  "Alerts": 1060,
  "Modal": 1050,
  "OverlaySurfaces": 1040,
  "Toast": 1070,
  "default": 1,
}
`;

exports[`utils Layers 2`] = `
Object {
  "Alerts": 1060,
  "Modal": 1050,
  "OverlaySurfaces": 1040,
  "Toast": 1070,
  "default": 1,
}
`;
